src/math-translator.js

```diff
@@ -90,36 +133,14 @@ function translateNumbers(math, lang) {
 
     // These consts are used only here for manipulating thousand separator
     const placeholder = 'THSEP';
-    const USThousandSeparatorRegex =
+    const thousandSeparatorRegex =
         new RegExp(`([0-9])${placeholder}([0-9])(?=[0-9]{2})`, 'g');
     const thousandSeparatorLocales = [].concat(
             MATH_RULES_LOCALES.THOUSAND_SEP_AS_THIN_SPACE,
             MATH_RULES_LOCALES.THOUSAND_SEP_AS_DOT,
             MATH_RULES_LOCALES.NO_THOUSAND_SEP);
 
-    // Definition of regex for decimal numbers
-    // We need to allow for strings like '\\greenD{3}.\\blue{1}' or
-    // repeating decimals like '1/3 = 0.\\overline{3}'
-    //
-    // Colors currently used in KA strings taken from KaTeX definitions, see:
-    // https://github.com/KaTeX/KaTeX/blob/master/src/macros.js
-    //
-    // \\overline is handled elsewhere since it appears only
-    // on the right side of the decimal point
-    //
-    // These colors are appended by optional [A-Z]? to match all definitions
-    // from KaTeX. This will form a superset of actually defined colors,
-    // but that hardly matters here and is more future-proof if new colors
-    // were defined at some point
-    const katexColorMacros = ['blue', 'gold', 'gray', 'mint', 'green', 'red',
-         'maroon', 'orange', 'pink', 'purple', 'teal', 'kaBlue', 'kaGreen']
-         .join('|');
-
-    const integerPart = `[0-9]+|\\\\(?:${katexColorMacros})[A-Z]?\\{[0-9]+\\}`;
-    const decPart =
-       `[0-9]+|\\\\(?:overline|${katexColorMacros})[A-Z]?\\{[0-9]+\\}`;
-    const decimalNumberRegex =
-      new RegExp(`(${integerPart})\\.(${decPart})`, 'g');
+    const decimalNumberRegex = new RegExp(getDecNumberRegexString('en'), 'g');
```

Comment:
I like how you've extracted this into a function, but I'm wondering if over time we'll lose track of how many capture groups these functions return?

Reply:
Added a comment to function JSDoc.